playbooks/aws/openshift-cluster/build_ami.yml

```diff
@@ -3,67 +3,49 @@
   connection: local
   gather_facts: no
   tasks:
-  - name: get the necessary vars for ami building
-    include_vars: vars.yml
-
-  - name: create a vpc with the name <clusterid>
-    include_role:
-      name: openshift_aws_vpc
-    vars:
-      r_openshift_aws_vpc_clusterid: "{{ provision.clusterid }}"
-      r_openshift_aws_vpc_cidr: "{{ provision.vpc.cidr }}"
-      r_openshift_aws_vpc_subnets: "{{ provision.vpc.subnets }}"
-      r_openshift_aws_vpc_region: "{{ provision.region }}"
-      r_openshift_aws_vpc_tags: "{{ provision.vpc.tags }}"
-      r_openshift_aws_vpc_name: "{{ provision.vpc.name | default(provision.clusterid) }}"
-
-  - name: create aws ssh keypair
+  - name: Require openshift_aws_base_ami
+    fail:
+      msg: "A base AMI is required for AMI building.  Please ensure  `openshift_aws_base_ami` is defined."
+    when: openshift_aws_base_ami is undefined
+
+  - name: "Alert user to variables needed and their values - {{ item.name }}"
+    debug:
+      msg: "{{ item.msg }}"
+    with_items:
+    - name: openshift_aws_clusterid
+      msg: "openshift_aws_clusterid={{ openshift_aws_clusterid | default('default') }}"
+    - name: openshift_aws_region
+      msg: "openshift_aws_region={{ openshift_aws_region | default('us-east-1') }}"
+
+  - name: create an instance and prepare for ami
     include_role:
-      name: openshift_aws_ssh_keys
+      name: openshift_aws
+      tasks_from: build_ami.yml
     vars:
-      r_openshift_aws_ssh_keys_users: "{{ provision.instance_users }}"
-      r_openshift_aws_ssh_keys_region: "{{ provision.region }}"
+      openshift_aws_node_group_type: compute
 
-  - name: fetch the default subnet id
-    ec2_vpc_subnet_facts:
-      region: "{{ provision.region }}"
+  - name: fetch newly created instances
+    ec2_remote_facts:
+      region: "{{ openshift_aws_region | default('us-east-1') }}"
       filters:
-        "tag:Name": "{{ provision.vpc.subnets[provision.region][0].az }}"
-    register: subnetout
-
-  - name: create instance for ami creation
-    ec2:
-      assign_public_ip: yes
-      region: "{{ provision.region }}"
-      key_name: "{{ provision.node_group_config.ssh_key_name }}"
-      group: "{{ provision.clusterid }}"
-      instance_type: m4.xlarge
-      vpc_subnet_id: "{{ subnetout.subnets[0].id }}"
-      image: "{{ provision.build.base_image }}"
-      volumes:
-      - device_name: /dev/sdb
-        volume_type: gp2
-        volume_size: 100
-        delete_on_termination: true
-      wait: yes
-      exact_count: 1
-      count_tag:
-        Name: ami_base
-      instance_tags:
-        Name: ami_base
-    register: amibase
+        "tag:Name": "{{ openshift_aws_ami_base_name | default('ami_base') }}"
```

Comment:
I'm pretty sure this should be "openshift_aws_base_ami_name" not "openshift_aws_ami_base_name"